File: test_hl7_send.py
#!/usr/bin/env python3
"""Send test HL7 message to Test Interface3"""
import socket
import time

# HL7 message
hl7_message = """MSH|^~\\&|SendingApp|SendingFac|ReceivingApp|ReceivingFac|20251116103500||ADT^A01|MSG00005|P|2.5
EVN||20251116103500
PID|||67890^^^MRN||TestUser^Mike^B||19851215|M|||789 Oak St^^CityName^CA^90001^USA||(555)555-7777|||M||9876543|456-78-9012
PV1||E|ER^201^01||||DOC789^Brown^Alice^N^MD^^^L|||ORTHO||||U|||DOC789^Brown^Alice^N^MD^^^L||VIS789|INS3||||||||||||||||||||||||||20251116103500
"""

# MLLP framing
start_block = b'\x0B'
end_block = b'\x1C\x0D'
message = start_block + hl7_message.encode('utf-8').replace(b'\n', b'\r') + end_block

# Send to Test Interface3 on port 6663
try:
    client = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
    client.settimeout(5)
    client.connect(('localhost', 6663))

    print("Connected to Test Interface3 on port 6663")
    client.sendall(message)
    print(f"Sent {len(message)} bytes")

    # Wait for ACK
    response = client.recv(1024)
    print(f"Received response ({len(response)} bytes):")
    print(response.decode('utf-8', errors='ignore'))

    client.close()
    print("\nTest message sent successfully!")
except Exception as e:
    print(f"Error: {e}")

File: tests/integration/test_hl7_send.js
const net = require('net');

// MLLP protocol wrapper function
function wrapMLLP(message) {
    const START_BLOCK = '\x0B'; // VT (Vertical Tab)
    const END_BLOCK = '\x1C';   // FS (File Separator)
    const CARRIAGE_RETURN = '\x0D'; // CR (Carriage Return)

    return START_BLOCK + message + END_BLOCK + CARRIAGE_RETURN;
}

// HL7 Test Message
const hl7Message = `MSH|^~\\&|SendingApp|SendingFacility|ReceivingApp|ReceivingFacility|20250921124500||ADT^A01^ADT_A01|MSG001|P|2.5\r` +
`EVN|A01|20250921124500|||^ADMIN^USER\r` +
`PID|1||123456^^^MRN||Doe^John^A||19850315|M|||123 Main St^^Anytown^NY^12345^USA||555-123-4567|||S||123456789|987-65-4321\r` +
`NK1|1|Doe^Jane^M|SPO|||||555-987-6543\r` +
`PV1|1|I|ICU^101^01||||DOC001^Doctor^Primary|||SUR||||1|A|||||||||||||||||||||||20250921124500`;

console.log('🚀 Sending HL7 message to TCP interface on port 6661...');
console.log('📋 Message Type: ADT^A01 (Admission)');
console.log('🏥 Patient: John Doe (MRN: 123456)');

const client = net.createConnection({ port: 6661, host: 'localhost' }, () => {
    console.log('✅ Connected to TCP interface');

    // Send MLLP-wrapped message
    const mllpMessage = wrapMLLP(hl7Message);
    client.write(mllpMessage);

    console.log('📤 HL7 message sent via MLLP protocol');
});

client.on('data', (data) => {
    console.log('📥 Received acknowledgment:', data.toString());
});

client.on('end', () => {
    console.log('🔚 Connection closed by server');
});

client.on('error', (err) => {
    console.error('❌ Connection error:', err.message);
});

// Close connection after 2 seconds
setTimeout(() => {
    client.end();
    console.log('✅ Test completed');
    process.exit(0);
}, 2000);